app/grandchallenge/reader_studies/tasks.py

```diff
@@ -79,17 +84,19 @@ def add_image_to_display_set(
             interface=interface, image=image
         ).first()
         if civ is None:
-            civ = ComponentInterfaceValue(interface=interface)
-        civ.image = image
-        try:
-            civ.full_clean()
-        except ValidationError as e:
-            upload_session.status = RawImageUploadSession.FAILURE
-            upload_session.error_message = e.message
-            upload_session.save()
+            civ = ComponentInterfaceValue.objects.create(
+                interface=interface, image=image
+            )
+            try:
+                civ.full_clean()
+            except ValidationError as e:
+                # this should only happen for new uploads
+                upload_session.status = RawImageUploadSession.FAILURE
+                upload_session.error_message = e.message
+                upload_session.save()
         else:
             civ.save()
```

Comment:
Why is saving the CIV reqired here now? It shouldn't be now that you're using create.

Reply:
You're right, it's not necessary.